src/Microsoft.AspNetCore.Hosting/WebHostExtensions.cs

```diff
@@ -15,11 +15,86 @@ namespace Microsoft.AspNetCore.Hosting
     public static class WebHostExtensions
     {
         /// <summary>
+        /// Blocks the calling thread until the web application shuts down.
+        /// </summary>
+        /// <param name="host">The <see cref="IWebHost"/> to wait for shutdown</param>
+        public static void WaitForShutdown(this IWebHost host)
+        {
+            WaitForSystemShutdown(host, (token, message) => host.WaitForShutdown(token, message));
+        }
+
+        /// <summary>
+        /// Blocks the calling thread until the web application shuts down.
+        /// </summary>
+        /// <param name="host">The <see cref="IWebHost"/> to wait for shutdown</param>
+        /// <param name="token">The token to trigger shutdown.</param>
+        public static void WaitForShutdown(this IWebHost host, CancellationToken token)
+        {
+            host.WaitForShutdown(token, shutdownMessage: null);
```

Comment:
Should this dispose the host? Part of the API is about simplicity but it's also generally useful outside of that use case.